### PR TITLE
Fixes #24154: The UI is barely functional

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/24154

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Review by VinceMacBuche — APPROVED** [review]

File: datasources/src/main/elm/sources/Update.elm [hunk 1]
```diff
@@ -58,16 +59,35 @@ update msg model =
...
+          -- Update scenario
+          ShowDatasource _ (Just _) ->
+            ( { model | dataSources = if(List.any (\dts -> dts.id == d.id) listDatasources) then listDatasources else d :: listDatasources}
+            , Cmd.batch [saveDataSource model False d, successNotification ("Datasource '"++ d.name ++"' successfully updated") ]
+            )
+          _ ->
+            (model, Cmd.none)
+
+
+    OpenDeleteModal datasource ->
```
Comment (ElaadF) [inline datasources/src/main/elm/sources/Update.elm:83]: This is the logic for open/close deletion

File: datasources/src/main/elm/sources/View.elm [hunk 2]
```diff
@@ -55,7 +55,9 @@ view model =
             h1 [] [text "Node properties data sources"]
           , p [] [
               text """Nodes have properties that can be used to create groups or in techniques and directives parameters.
-              These properties are b [] [key/value] pairs, where values can be a simple string or a well formed JSON document."""
+              These properties are"""
+              , b [] [text " key/value pairs "]
```
Comment (ElaadF) [inline datasources/src/main/elm/sources/View.elm:59]: Just a typo

File: datasources/src/main/elm/sources/Update.elm [hunk 3]
```diff
@@ -58,16 +59,35 @@ update msg model =
...
+      let
+         listDatasources = model.dataSources
+      in
+        case model.mode of
+          -- Creation scenario
+          ShowDatasource _ Nothing ->
+            let
+              dataSource = {d | id = if String.isEmpty d.id then d.name else d.id }
+            in
+              ( { model | dataSources = dataSource :: listDatasources }
```
Comment (ElaadF) [inline datasources/src/main/elm/sources/Update.elm:71]: When we create a Datasource, the left menu was not updated withe newly created datasource

**Comment by ElaadF** [comment]
OK, merging this PR

**Pull request merged**